idaes/surrogate/pysmo/sampling.py

```diff
@@ -314,6 +314,100 @@ def data_sequencing(self, no_samples, prime_base):
+
+            if not set_of_labels: 
+                # Empty list assumes one output in final column - all other variables treated as inputs
+                print('No column information provided. All except last column will be considered as x variables.')
+                self.data = data_input.values
+                self.data_headers = data_input.columns.values.tolist()
+                self.x_data = self.data[:, :-1]
+                self.data_headers_xvars = self.data_headers[:-1] # Last column is y
+            elif not set(set_of_labels).issubset(data_input.columns):
+                raise IndexError('Invalid column names in "xlabels" or "ylabels". ')
+            elif not ylabels:
+                # When only xlabels are supplied, assume all non-inputs are outputs
+                self.x_data = data_input.filter(xlabels).values
+                self.data_headers_xvars = xlabels
+                non_x_vars = [k for k in data_input.columns if k not in xlabels]
+                y_data = data_input.filter(non_x_vars).values
+                self.data_headers = self.data_headers_xvars + non_x_vars
+                self.data = np.concatenate((self.x_data, y_data), axis=1)
+            elif not xlabels:
+                # When only ylabels are supplied, assume all non-outputs are inputs
+                xvars = [k for k in data_input.columns if k not in ylabels]
+                self.x_data = data_input.filter(xvars).values
+                self.data_headers_xvars = xvars
+                y_data = data_input.filter(ylabels).values
+                self.data_headers = self.data_headers_xvars + ylabels
+                self.data = np.concatenate((self.x_data, y_data), axis=1)
+            else:
+                # Only provided columns are considered - any column not in xlabels and ylabels is dropped completely.
+                self.x_data = data_input.filter(xlabels).values
+                self.data_headers = set_of_labels
+                self.data_headers_xvars = xlabels
+                self.data = data_input.filter(set_of_labels).values
+
+
+        elif isinstance(data_input, np.ndarray):
+            self.df_flag = False 
+
+            if xlabels is None:
+                xlabels = []
+            if ylabels is None:
+                ylabels = []
+            set_of_labels = xlabels + ylabels
+
+            if not set_of_labels: 
+                # Empty list assumes one output in final column
+                print('No column information provided. All except last column will be considered as x variables.')
+                self.data = data_input
+                self.data_headers = [i for i in range(0, data_input.shape[1])]
+                self.x_data = self.data[:, :-1]
+                self.data_headers_xvars = self.data_headers[:-1] # Last column is y
+            elif not set(set_of_labels).issubset(range(0, data_input.shape[1])):
+                raise IndexError('Invalid column names in "xlabels" or "ylabels". ')
```

Comment:
should say which names are invalid

Reply:
For now PySMO uses print statements throughout the entire package. I'd like to keep it consistent for now.

Changing all the print statements in PySMO to logs will a separate undertaking and will be addressed in a separate PR later.

Reply:
Done.